### ISSUE TITLE
language model，how to customize template operations or skip default operations. 自定义了 tokenizer 的输出，导致 template 相关逻辑错误，怎么可以跳过 template 或者自定义 template 操作？

### ISSUE THREAD

**Issue by zhejia** [modelscope/ms-swift#6526]
自定义了 tokenizer 的输出，增加了token的相关逻辑操作，tokenizer 会返回 token_ids、attention_mask、position_ids、labels、以及一些额外的自定义字段。而 template 相关处理逻辑会导致拼接错误，怎么可以跳过 template 或者自定义 template 操作？

期望的是 tokenizer 的处理结果能直接传入模型，可以怎么做？

**Comment by zhejia** [comment]
自定义 dataset 的 preprocess_func 能解决这个问题吗？
目前我的流程看起来是这样的执行的， tokenizer -> preprocess（我写在了 tokenizer 内部）-> template -> model
自定义 dataset 的 preprocess_func 能实现 tokenizer -> template -> preprocess -> model 这样的流程吗？
